WordPress/Sniffs/VIP/ValidatedSanitizedInputSniff.php

```diff
@@ -45,6 +45,26 @@ public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr )
 	{
 		$tokens = $phpcsFile->getTokens();
 
+		/*
+		 * Quoting 'http://php.net/manual/en/reserved.variables.server.php' :
+		 *
+		 * There is no guarantee that every web server will provide any of these;
+		 * servers may omit some, or provide others not listed here.
+		 *
+		 * That said, a large number of these variables are accounted for in
+		 * the » CGI/1.1 specification, so you should be able to expect those.
+		 * http://www.faqs.org/rfcs/rfc3875.html
+		 *
+		 * @see https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/191
+		 */
+		$server_vars = array(
+			'AUTH_TYPE', 'CONTENT_LENGTH', 'CONTENT_TYPE', 'GATEWAY_INTERFACE',
+			'PATH_INFO', 'PATH_TRANSLATED', 'QUERY_STRING', 'REMOTE_ADDR',
+			'REMOTE_HOST', 'REMOTE_IDENT', 'REMOTE_USER', 'REQUEST_METHOD',
+			'SCRIPT_NAME', 'SERVER_NAME', 'SERVER_PORT', 'SERVER_PROTOCOL',
+			'SERVER_SOFTWARE',
+		);
```

Comment:
Related: 
- [[33775]](https://core.trac.wordpress.org/changeset/33775) - Avoid PHP notices in `redirect_canonical()` and `_wp_menu_item_classes_by_context()` if `$_SERVER['HTTP_HOST']` is not set.
- [[32839]](https://core.trac.wordpress.org/changeset/32839) - Make sure `$_SERVER['SERVER_NAME']` is set whenever `wp_mail()` is called in PHPUnit tests.
- [[32785]](https://core.trac.wordpress.org/changeset/32785) - Set `$_SERVER['REQUEST_METHOD']` for unit tests.

In other words, WordPress assumes that `REQUEST_METHOD` and `SEVER_NAME` will always be set, and monkey-patches them when running the PHPUnit tests.

It doesn't assume that `HTTP_HOST` is set, and it is not in this list either.